src/main/webapp/script.js

```diff
@@ -1,28 +1,139 @@
-// Copyright 2020 Google LLC
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     https://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Do you like guacamole?', 'I went to six different schools', 'Are you also a cat person?', '你好呀朋友'];
-
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
-
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+
+//habit list item array
+let habitItems = [];
+
+//create a new habit item based on the text from input
+//add to habitItems list
+function addHabit(text) {
+    const habit = {
+      text,
+      checked: false,
+      id: Date.now(),
+    };
+  
+    habitItems.push(habit);
+    renderHabit(habit);
+    console.log(habitItems);
+  }
+
+//select the from element 
+form = document.querySelector("#js-form");
+if(form)
+{
+    console.log("get form");
+
+//submit event listener
+//when a new habit sumbitted, it will be added to habitItems list
+    form.addEventListener('submit', event => {
+        event.preventDefault();
+        console.log("submitted");
+        //prevent page from refreshing when form submitted
+        event.preventDefault();
+    
+        //select text input
+        input = document.querySelector('#js-habit-input');
+    
+        //process input
+        //remove whitespace
+        text = input.value.trim();
+        //add to habitItems list
+        if (text !== ''){
+            addHabit(text);
+            console.log("add habit");
+            input.value = '';
+            input.focus();
+        }
+    });
+}
+//render habits on the screen
+function renderHabit(habit){
+    //select list item in html
```

Comment:
Nit: Don't think the comment is necessary since your variables are named very well.